homeassistant/components/usb/__init__.py

```diff
@@ -410,6 +459,19 @@ async def _async_process_ports(self, ports: Sequence[ListPortInfo]) -> None:
                 )
             ]
 
+        unique_devices = set(usb_devices)
+        added_devices = unique_devices - self._last_processed_devices
+        removed_devices = self._last_processed_devices - unique_devices
+        self._last_processed_devices = unique_devices
+
+        _LOGGER.debug(
+            "Added devices: %r, removed devices: %r", added_devices, removed_devices
+        )
+
+        if added_devices or removed_devices:
+            for callback in self._port_event_callbacks:
```

Comment:
If the callback can remove itself, you'll need to make a copy here to avoid iteration during change